pkg/cloud/cloud_suite_test.go

```diff
@@ -17,13 +17,76 @@ limitations under the License.
 package cloud_test
 
 import (
+	"os"
+	"strings"
 	"testing"
 
+	"github.com/apache/cloudstack-go/v2/cloudstack"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"sigs.k8s.io/cluster-api-provider-cloudstack/pkg/cloud"
+	"sigs.k8s.io/cluster-api-provider-cloudstack/test/dummies"
+	"sigs.k8s.io/cluster-api-provider-cloudstack/test/helpers"
+)
+
+var (
+	realCloudClient cloud.Client
+	client          cloud.Client
+	realCSClient    *cloudstack.CloudStackClient
+	testDomainPath  string // Needed in before and in after suite.
 )
 
 func TestCloud(t *testing.T) {
 	RegisterFailHandler(Fail)
+	BeforeSuite(func() {
+		suiteConfig, _ := GinkgoConfiguration()
+		if !strings.Contains(suiteConfig.LabelFilter, "!integ") { // Skip if integ tests are filtered out.
```

Comment:
If integ tests are not filtered out, then the test suite also creates a real client in a temporary account/domain that's wiped out at the end of testing.